core/api.go

```diff
@@ -316,6 +317,22 @@ func blockchainIDHandler(handler http.Handler, blockchainID string) http.Handler
 	})
 }
 
+// loggingHandler pulls out request data and adds it to the request's
+// logging context.
+func loggingHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		ctx = log.AddPrefixkv(ctx, "path", req.URL.Path)
+		if userAgent := req.UserAgent(); userAgent != "" {
+			ctx = log.AddPrefixkv(ctx, "useragent", userAgent)
```

Comment:
With our current logging, I think that'd actually be more verbose. Most of our logging is only when an error occurs, but logging here would log the user agent on every request.